src/components/top-header/top-header.tsx

```diff
@@ -38,21 +44,42 @@ export function TopHeader() {
         <Grid.Col span="auto">
           <Group position="right" spacing="xl">
             <Link href="/">
-              <HeaderButton text="Home"></HeaderButton>
+              <HeaderButton>Home</HeaderButton>
             </Link>
             <MenuHeader />
             <Link href="/location" passHref>
-              <HeaderButton text="Location"></HeaderButton>
+              <HeaderButton>Location</HeaderButton>
             </Link>
             <Link href="/contact" passHref>
-              <HeaderButton text="Contact Us"></HeaderButton>
+              <HeaderButton>Contact Us</HeaderButton>
             </Link>
             <Link href="/about">
-              <HeaderButton text="About"></HeaderButton>
-            </Link>
-            <Link href="/admin">
-              <HeaderButton text="Admin"></HeaderButton>
+              <HeaderButton>About</HeaderButton>
             </Link>
+            {session?.user.isAdmin ? (
+              <Link href="/admin">
+                <HeaderButton>Admin</HeaderButton>
+              </Link>
+            ) : null}
+
+            {pathname !== '/auth/signin' ? (
+              session?.user ? (
+                <>
+                  <GeneralButton
+                    onClick={() => {
+                      signOut()
+                    }}
+                  >
+                    Sign Out
+                  </GeneralButton>
+                </>
+              ) : (
+                <GeneralButton onClick={() => signIn()}>Sign In</GeneralButton>
+              )
+            ) : null}
+            {session?.user ? (
+              <Avatar color={MainColors.RedStatic} src={session.user.image} size={40} radius={40} />
```

Comment:
![image](https://github.com/Jax-Donuts/Jax-website/assets/138041702/97c2c1a6-4c9b-435c-bb31-ea57665f9d49)
![image](https://github.com/Jax-Donuts/Jax-website/assets/138041702/2a21a2b2-70ca-400d-b15c-db77a2db4b49)
![image](https://github.com/Jax-Donuts/Jax-website/assets/138041702/982ec64e-0357-450d-87c0-00cccb910cb5)


I'm not sure if color is for the default profile icon but when signed into with a different email, icon is blue (even though that gmail has a profile pic)

Reply:
Im not sure how what I did fixed anything so Ill need you to double check this after I rerequest a review.